app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -0,0 +1,429 @@
+package com.github.sdpteam15.polyevents.helper
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.drawable.Drawable
+import android.util.Log
+import androidx.core.content.ContextCompat
+import com.github.sdpteam15.polyevents.R
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.*
+import kotlin.math.pow
+
+enum class PolygonAction {
+    RIGHT,
+    DOWN,
+    DIAG,
+    MOVE,
+    ROTATE
+}
+
+object GoogleMapHelper {
+    var context: Context? = null
+    var map: GoogleMap? = null
```

Comment:
Isn't the GoogleMap object guaranteed to be non null ? See https://developers.google.com/android/reference/com/google/android/gms/maps/MapFragment#public-void-getmapasync-onmapreadycallback-callback